src/Symfony/Component/HttpKernel/HttpCache/Store.php

```diff
@@ -325,10 +325,13 @@ private function getMetadata($key)
      */
     public function purge($url)
     {
-        $http = preg_replace('#^https#', 'http', $url);
-        $https = preg_replace('#^http#', 'https', $url);
+        $http = preg_replace('#^https:#', 'http:', $url);
+        $https = preg_replace('#^http:#', 'https:', $url);
 
-        return $this->doPurge($http) || $this->doPurge($https);
+        $purgedHttp = $this->doPurge($http);
+        $purgedHttps = $this->doPurge($https);
+
+        return $purgedHttp || $purgedHttps;
```

Comment:
That was the problem. Both versions of the cache entry – http AND https – should get purged. Before this change only the http version was purged.